specification/connector/_types/Connector.ts

```diff
@@ -0,0 +1,241 @@
+  tooltip: string | null
+  type: ConnectorFieldType
+  ui_restrictions: string[]
+  validations: Validation[]
+  value: string | number | boolean | null
+}
+
+export type ConnectorConfiguration = Dictionary<
+  string,
+  ConnectorConfigProperties | ConnectorConfigCategoryProperties | null
+>
+
+export interface ConnectorScheduling {
+  enabled: boolean
+  interval: string // interval has crontab syntax
+}
+
+interface CustomScheduling {
+  configuration_overrides: Dictionary<string, UserDefinedValue>
+  enabled: boolean
+  interval: string
+  last_synced: string | null
+  name: string
+}
+
+export type ConnectorCustomScheduling = Dictionary<
+  string,
+  CustomScheduling | null
+>
+
+enum ConnectorStatus {
+  CREATED = 'created',
+  NEEDS_CONFIGURATION = 'needs_configuration',
+  CONFIGURED = 'configured',
+  CONNECTED = 'connected',
+  ERROR = 'error'
+}
+
+export enum SyncStatus {
+  CANCELING = 'canceling',
+  CANCELED = 'canceled',
+  COMPLETED = 'completed',
+  ERROR = 'error',
+  IN_PROGRESS = 'in_progress',
+  PENDING = 'pending',
+  SUSPENDED = 'suspended'
+}
+
+export interface IngestPipelineParams {
+  extract_binary_content: boolean
+  name: string
+  reduce_whitespace: boolean
+  run_ml_inference: boolean
+}
+
+type FilteringPolicy = 'exclude' | 'include'
+
+type FilteringRuleRule =
+  | 'contains'
+  | 'ends_with'
+  | 'equals'
+  | '>'
+  | '<'
+  | 'regex'
+  | 'starts_with'
+
+interface FilteringRule {
+  created_at: string
+  field: string
+  id: string
+  order: number
+  policy: FilteringPolicy
+  rule: FilteringRuleRule
+  updated_at: string
+  value: string
+}
+
+interface FilteringValidation {
+  ids: string[]
+  messages: string[]
+}
+
+enum FilteringValidationState {
+  EDITED = 'edited',
+  INVALID = 'invalid',
+  VALID = 'valid'
+}
+
+interface FilteringAdvancedSnippet {
+  created_at: string
+  updated_at: string
+  value: Dictionary<string, UserDefinedValue>
+}
+
+interface FilteringRulesValidation {
+  errors: FilteringValidation[]
+  state: FilteringValidationState
+}
+
+interface FilteringRules {
+  advanced_snippet: FilteringAdvancedSnippet
+  rules: FilteringRule[]
+  validation: FilteringRulesValidation
+}
+
+export interface FilteringConfig {
+  active: FilteringRules
+  domain: string
+  draft: FilteringRules
+}
+
+enum FeatureName {
+  FILTERING_ADVANCED_CONFIG = 'filtering_advanced_config',
+  FILTERING_RULES = 'filtering_rules',
+  DOCUMENT_LEVEL_SECURITY = 'document_level_security',
+  INCREMENTAL_SYNC = 'incremental_sync',
+  SYNC_RULES = 'sync_rules'
+}
+
+interface FeatureEnabled {
+  enabled: boolean
+}
+
+interface SyncRulesFeature {
+  advanced?: FeatureEnabled
+  basic?: FeatureEnabled
+}
+
+export interface ConnectorFeatures {
+  [FeatureName.DOCUMENT_LEVEL_SECURITY]?: FeatureEnabled
+  [FeatureName.FILTERING_ADVANCED_CONFIG]?: boolean
+  [FeatureName.FILTERING_RULES]?: boolean
+  [FeatureName.INCREMENTAL_SYNC]?: FeatureEnabled
+  [FeatureName.SYNC_RULES]?: SyncRulesFeature
+}
```

Comment:
```suggestion
export interface ConnectorFeatures {
  document_level_security?: FeatureEnabled
  filtering_advanced_config?: boolean
  filtering_rules?: boolean
  incremental_sync?: FeatureEnabled
  sync_rules?: SyncRulesFeature
}
```

That mechanism isn't implemented in the compiler, this results in an object with the properties called `[FeatureName.DOCUMENT_LEVEL_SECURITY]` which isn't what you wanted I presume.

These properties should be name directly from their counterparts in `FeatureName` and the enum removed.